### PR TITLE
Add example for advanced cloud variable types in MicroPython

### PR DESCRIPTION
**Pull request by sebromero**
## What This PR Changes
- Adds an example on how to use advanced cloud variable types in MicroPython

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/arduino-cloud/01.guides/04.micropython/content.md [hunk 1]
```diff
@@ -214,6 +215,204 @@ if __name__ == "__main__":
 
 Open Arduino Lab for MicroPython and connect to your board. Pasting the above code and run the script. Then open your Arduino Cloud dashboard. You should see the registered "ledSwitch" and "led" widgets. Toggle the "ledSwitch", and the LED on your Arduino board should light up accordingly. The state of the "led" variable should also change, mirroring the state of the physical LED.
 
+## Using Advanced Cloud Variables
+
+To use variables in Arduino IoT Cloud that are not of a basic type, you can set them up like you would do with any other variable. For example, to control a colored light such as the on-board RGB led on some Arduino boards, you can create a variable of type `CloudColoredLight`. ![Variable creation in Arduino Cloud](./assets/colored-light-variable.png)
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.guides/04.micropython/content.md:220]:
```suggestion
To use variables in Arduino Cloud that are not of a basic type, you can set them up like you would do with any other variable. For example, to control a colored light such as the on-board RGB led on some Arduino boards, you can create a variable of type `CloudColoredLight`. 

![Variable creation in Arduino Cloud](./assets/colored-light-variable.png)

```

File: content/arduino-cloud/01.guides/04.micropython/content.md [hunk 2]
```diff
@@ -214,6 +215,204 @@ if __name__ == "__main__":
...
+        rgb_value = convert_hs_to_rgb(light.hue, light.sat, light.bri)
+        set_leds_from_rgb(rgb_value)
+    else:
+        set_leds_from_rgb((0, 0, 0)) # Turn LEDs off
+```
+
+Here is the complete code to try it out:
+
+```python
+# This file is part of the Python Arduino IoT Cloud.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.guides/04.micropython/content.md:301]:
```suggestion
# This file is part of the Python Arduino Cloud.

```

File: content/arduino-cloud/01.guides/04.micropython/content.md [hunk 3]
```diff
@@ -214,6 +215,204 @@ if __name__ == "__main__":
...
+    logging.basicConfig(
+        datefmt="%H:%M:%S",
+        format="%(asctime)s.%(msecs)03d %(message)s",
+        level=logging.INFO,
+    )
+    
+    # NOTE: Add networking code here or in boot.py
+    wifi_connect()
+    
+    # Create a client object to connect to the Arduino IoT cloud.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.guides/04.micropython/content.md:405]:
```suggestion
    # Create a client object to connect to the Arduino Cloud.

```

File: content/arduino-cloud/01.guides/04.micropython/content.md [hunk 4]
```diff
@@ -214,6 +215,204 @@ if __name__ == "__main__":
...
+    # For MicroPython, the key and cert files must be stored in DER format on the filesystem.
+    # Alternatively, a username and password can be used to authenticate:
+    client = ArduinoCloudClient(device_id=DEVICE_ID, username=DEVICE_ID, password=CLOUD_PASSWORD)
+    client.register(ColoredLight("light", swi=True, on_write=on_colored_light_changed))
+    client.register("led", value=None)
+
+    # Start the Arduino IoT cloud client.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.guides/04.micropython/content.md:412]:
```suggestion
    # Start the Arduino Cloud client.

```

**Review by jhansson-ard — COMMENTED** [review]
Some updates to the Arduino Cloud naming, otherwise it looks good from a wording perspective from my side!

**Comment by sebromero** [comment]
@jhansson-ard Makes sense, I thought I had replaced them already, but looks like I didn't. Thanks for the suggestions!

File: content/arduino-cloud/01.guides/04.micropython/content.md [hunk 5]
```diff
@@ -214,6 +215,210 @@ if __name__ == "__main__":
...
+    light_enabled = light.swi
+    
+    if light_enabled:
+        rgb_value = convert_hs_to_rgb(light.hue, light.sat, light.bri)
+        set_leds_from_rgb(rgb_value)
+    else:
+        set_leds_from_rgb((0, 0, 0)) # Turn LEDs off
+```
+
+Here is the complete code to try it out:
```
Comment (karlsoderby) [inline content/arduino-cloud/01.guides/04.micropython/content.md:301]:
```suggestion
### Full Code Example

Here is the complete code to try it out:

```

Added a level 3 header just to separate the previous code snippets

**Review by karlsoderby — APPROVED** [review]
LGTM!

**Review by jhansson-ard — APPROVED** [review]

**Pull request merged**